From a4052a6c42bd60020590c528afd88ff7e05d7472 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Wed, 29 Feb 2012 15:18:42 +0800
Subject: [PATCH 245/635] ENGR00175702 enable the wm8958 in sabresd

The name has been changed and there is only one mic

Signed-off-by: b02247 <b02247@freescale.com>
---
 alsa/alsa_imx.cpp | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/alsa/alsa_imx.cpp b/alsa/alsa_imx.cpp
index a54d0be..36ab78c 100755
--- a/alsa/alsa_imx.cpp
+++ b/alsa/alsa_imx.cpp
@@ -38,7 +38,7 @@
 
 #ifdef  BOARD_IS_SABRELITE
 #define AUDIOCARD_DEVICE_SGTL5000_HIFI "HiFi sgtl5000-0"
-#define AUDIOCARD_DEVICE_WM8958_HIFI "WM8994 HiFi WM8994 AIF1-0"
+#define AUDIOCARD_DEVICE_WM8958_HIFI "HiFi wm8994-aif1-0"
 #define AUDIOCARD_DEVICE_WM8958_VOICE "WM8994 Voice WM8994 AIF2-1"
 #define AUDIOCARD_DEVICE_WM8958_BT "WM8994 BT WM8994 AIF3-2"
 #define AUDIOCARD_DEVICE_HDMI "IMX HDMI TX mxc-hdmi-soc-0"
@@ -800,21 +800,21 @@ void setDefaultControls(uint32_t devices, int mode, const char *cardname)
                     ctl->set("AIF1ADC1L Mixer ADC/DMIC Switch", 1, 0);
 
             }else if(devices & AudioSystem::DEVICE_IN_BUILTIN_MIC ){
-                ctl->set("IN1L Switch", 0, 0);
-                ctl->set("IN1R Switch", 1, 0);
-                ctl->set("IN1R Volume", 27, 0);
-                ctl->set("MIXINR IN1R Switch", 1, 0);
-                ctl->set("MIXINR IN1R Volume", 1, 0);
-                ctl->set("IN1R PGA IN1RN Switch", 1, 0);
-                ctl->set("IN1R PGA IN1RP Switch", 1, 0);
-                ctl->set("ADCR Mux", 0, 0);   /* 0: ADC  1: DMIC */
+                ctl->set("IN1R Switch", 0, 0);
+                ctl->set("IN1L Switch", 1, 0);
+                ctl->set("IN1L Volume", 27, 0);
+                ctl->set("MIXINL IN1L Switch", 1, 0);
+                ctl->set("MIXINL IN1L Volume", 1, 0);
+                ctl->set("IN1L PGA IN1LN Switch", 1, 0);
+                ctl->set("IN1L PGA IN1LP Switch", 1, 0);
+                ctl->set("ADCL Mux", 0, 0);   /* 0: ADC  1: DMIC */
                 if (mode == AudioSystem::MODE_IN_CALL)
                 {
-                    ctl->set("AIF2DAC2R Mixer Right Sidetone Switch", 1, 0);
-                    ctl->set("Right Sidetone", 0, 0);/*0: ADC/DMIC1, 1:DMIC2 */
+                    ctl->set("AIF2DAC2L Mixer Left Sidetone Switch", 1, 0);
+                    ctl->set("Left Sidetone", 0, 0);/*0: ADC/DMIC1, 1:DMIC2 */
                 }
                 else
-                    ctl->set("AIF1ADC1R Mixer ADC/DMIC Switch", 1, 0);
+                    ctl->set("AIF1ADC1L Mixer ADC/DMIC Switch", 1, 0);
             }
         }
     }
-- 
1.8.0

